### PR TITLE
Fix incorrect sorting in statement node contexts

### LINKED ISSUE #36146: Improve completion item sorting in already completed statements
**Description:**
Right now, LS sorts the completion items correctly for partially written statements. Ex:
```ballerina
public function main() returns error? {
    int val = get<cursor>
}

function getInt() returns int {
    return 1;
}

function getStr() returns string {
    return "";
}
```
At `<cursor>`, functions that return an int and int type variables will be sorted first. 

But if the user has already completed the statement similar to the below case, the sorting of completion items are not correct.
```ballerina
public function main() returns error? {
    int val = get<cursor>Str();
}

function getInt() returns int {
    return 1;
}

function getStr() returns string {
    return "";
}
```
In the above example, `getStr()` will be the 1st while `getInt()` comes 2nd. As per the variable declaration statement, user expect a function/var of type `int` to come first.

**Describe your problem(s)**
See description. 

**Describe your solution(s)**
Need to improve the LS `ContextTypeResolver` to derive the contextually expected type for already completed statements.

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

### PR DESCRIPTION
**Pull request by malinthar**
## Purpose
$subject by fixing incorrect context type resolution. 

Fixes #36146

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [x] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/ConstantDeclarationNodeContext.java [hunk 1]
```diff
@@ -105,8 +104,8 @@ public boolean onPreValidation(BallerinaCompletionContext context, ConstantDecla
     }
 
     @Override
-    protected List<LSCompletionItem> initializerContextCompletions(BallerinaCompletionContext context,
-                                                                   Node typeDesc, Node initializer) {
+    protected List<LSCompletionItem> initializerContextCompletions(BallerinaCompletionContext context,
```
Comment (mindula) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/ConstantDeclarationNodeContext.java:107]: Shall we remove the extra space after the comma?

Comment (mindula) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/ConstantDeclarationNodeContext.java:107]: Shall we remove the extra space after the comma?

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/ImplicitAnonymousFunctionExpressionNodeContext.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+import org.ballerinalang.langserver.commons.completion.LSCompletionException;
+import org.ballerinalang.langserver.commons.completion.LSCompletionItem;
+
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Completion provider for {@link io.ballerina.compiler.syntax.tree.ImplicitAnonymousFunctionExpressionNode} context.
+ *
+ * @since 2.0.0
```
Comment (TharushiJay) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/ImplicitAnonymousFunctionExpressionNodeContext.java:31]: Shall we update the version?

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/ConstantDeclarationNodeContext.java (hunk 1 above)
Comment (malinthar) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/ConstantDeclarationNodeContext.java:107]: If we do so the length of the line will be greater than 120 chars.

**Review by IMS94 — APPROVED** [review]

**Pull request merged**